docs/guides/runbooks_steps_firehydrant.md

```diff
@@ -0,0 +1,779 @@
+
+  steps {
+    name      = "Assign Commander Role"
+    action_id = data.firehydrant_runbook_action.firehydrant_assign_a_role.id
+
+    config = jsonencode({
+      role = {
+        label = "Commander"
+        value = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
+      }
+      user = {
+        label = "Incident Opener"
+        value = {
+          type = "incident_opener"
+        }
+      }
+    })
+
+    automatic = true
+    repeats   = false
+  }
+
+  steps {
+    name      = "Assign Ops Lead Role"
+    action_id = data.firehydrant_runbook_action.firehydrant_assign_a_role.id
+
+    config = jsonencode({
+      role = {
+        label = "Ops Lead"
+        value = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
+      }
+      user = {
+        label = "Example External Schedule"
+        value = {
+          type                 = "external_schedule"
+          external_schedule_id = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
+        }
+      }
+    })
+
+    automatic = true
+    repeats   = false
+  }
+
+  steps {
+    name      = "Assign Communication Role"
+    action_id = data.firehydrant_runbook_action.firehydrant_assign_a_role.id
+
+    config = jsonencode({
+      role = {
+        label = "Communication"
+        value = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
+      }
+      user = {
+        label = "Example User"
+        value = {
+          type    = "user"
+          user_id = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
+        }
+      }
+    })
+
+    automatic = true
+    repeats   = false
+  }
+}
+```
+
+### Assign a Role - Steps Argument Reference
+
+* `action_id` - (Required) The ID of the runbook action for the step.
+* `config` - (Required) JSON string representing the configuration settings for the step.
+  Use [Terraform's jsonencode](https://www.terraform.io/language/functions/jsonencode)
+  function so that [Terraform can guarantee valid JSON syntax](https://www.terraform.io/language/expressions/strings#generating-json-or-yaml).
+* `name` - (Required) The name of the step.
+* `automatic` - (Optional) Whether this step should automatically execute.
+* `repeats` - (Optional) Whether this step should repeat. Defaults to `false`.
+  When this value is `true`, `repeats_duration` _must_ be provided.
+* `repeats_duration` - (Optional) How often this step should repeat in ISO8601.
+  Example: PT10M [Format Spec](https://www.digi.com/resources/documentation/digidocs/90001437-13/reference/r_iso_8601_duration_format.htm)
+  This value _must_ be provided if `repeats` is `true`. This value _must not_ be provided if `repeats` is `false`.
+
+The `config` block supports:
+
+* `role` - (Required) The role to assign.
+* `user` - (Optional) The user to assign the role to.
+
+The `role` block supports:
+
+* `label` - (Required) The name of the role.
+* `value` - (Required) The ID of the role.
+
+The `user` block supports:
+
+* `label` - (Required) The name of the user.
+* `value` - (Required) The type and ID of the user.
```

Comment:
I went through a real rabbit hole here, but this actually needs to be JSON encoded. 
This is from the payload in the UI

```json
{
  user: {
    label: "Incident Opener",
    value: "{\"type\":\"incident_opener\"}"
  }
}
```

What's worse is we don't validate this input, resulting in a completely broken runbook

Reply:
does it work if you add jsonencode to the value, like this:
```
    config = jsonencode({
      role = {
        label = "Commander"
        value = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
      }
      user = {
        label = "Incident Opener"
        value = jsonencode({
          type = "incident_opener"
        })
      }
    })
```

Reply:
Still trying to uncorrupt my runbook 😓 
I should probably just delete it. Will try momentarily

Reply:
Yes. That fixes it, but also, it would be very nice to have that validation on the backend

Reply:
agreed!